lib/src/main/java/graphql/nadel/validation/NadelHydrationArgumentValidation.kt

```diff
@@ -0,0 +1,288 @@
+package graphql.nadel.validation
+
+import graphql.Scalars
+import graphql.nadel.dsl.RemoteArgumentDefinition
+import graphql.nadel.dsl.RemoteArgumentSource.SourceType.ObjectField
+import graphql.nadel.dsl.RemoteArgumentSource.SourceType.FieldArgument
+import graphql.nadel.dsl.UnderlyingServiceHydration
+import graphql.nadel.engine.util.isList
+import graphql.nadel.engine.util.isNonNull
+import graphql.nadel.engine.util.unwrapNonNull
+import graphql.nadel.engine.util.unwrapOne
+import graphql.schema.*
+
+internal class NadelHydrationArgumentValidation private constructor() {
+    companion object {
+        fun validateHydrationInputArg(
+                hydrationSourceFieldType: GraphQLType,
```

Comment:
```suggestion
                hydrationSourceFieldType: GraphQLOutputType,
```

Reply:
this isnt always a GraphQLOutputType as its sometimes an argument. Will keep as `GraphQLType` and rename `hydrationSourceFieldType` to something more appropriate